### DISCUSSION TITLE
Docker volumes/mounts

### DISCUSSION THREAD

**Discussion by Plevuus** [benjaminjonard/koillection#771] [Q&A]
Docker in general is still new to me and I'm on the learning curve. Installed Koillection 1.3.18 sometime in January using the docker compose file and I remember I struggled with the volumes and permissions. After some tinkering with the file and the volume paths it worked and it has been working ever since. I'm looking into it again and I now already have a better understanding of volumes and mounts and I figured out that my collection and mysql database are in mounts and not in volumes. I'm just wondering how it is supposed to be. Should it be mounts or volumes?

Edit:
It seems the issues all started with using snap docker.

**Comment by benjaminjonard** [comment]
I usually prefer bind mounts because I can easily access the files if needed.

I think Docker recommends using named volumes though, because it makes your docker stack more portable and it should prevent access rights issues.

There is quite a lot of topics concerning this on stack overflow and reddit, but the only consensus is that there is no absolute answer. You can use bind mounts or named volumes, depending on your needs. Koillection should work with both.

**Reply by Basserado** [reply]
I have a similar problem: I installed Koillection via Docker Hub through Docker Compose. Everything works fine, but I want to modify some of the application files. So I tried to bind mount the /var/www directory of the Koillection container to a local directory. But then I get the following error when I restart the container: "sh: 0: cannot open /var/www/koillection/docker/entrypoint.sh: No such file"
